File: agent-frontend/src/app/model/comment.ts
import { Company } from './company';
import { User } from './user';

export interface CommentInterface {
  id?: number;
  text: string;
  company:Company;
  user:User;
}
export class Comment implements CommentInterface {
  id?: number;
  text: string;
  company:Company;
  user:User;

  constructor(obj: CommentInterface) {
      this.id = obj.id;
      this.text = obj.text;
      this.company = obj.company;
      this.user=obj.user;
  }

}

File: Agent/src/main/java/com/example/Agent/model/Comment.java
package com.example.Agent.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("comments")
public class Comment {
	@Id
	private Long id;
	private String text;
	private Company company;
	private User user;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getText() {
		return text;
	}
	public void setText(String text) {
		this.text = text;
	}
	public Company getCompany() {
		return company;
	}
	public void setCompany(Company company) {
		this.company = company;
	}
	
	public User getUser() {
		return user;
	}
	public void setUser(User user) {
		this.user = user;
	}
	
	
	public Comment(Long id, String text, Company company, User user) {
		super();
		this.id = id;
		this.text = text;
		this.company = company;
		this.user = user;
	}
	public Comment() {}
}
